src/doc/src/reference/environment-variables.md

```diff
@@ -257,6 +257,12 @@ corresponding environment variable is set to the empty string, `""`.
   where integration tests or benchmarks are free to put any data needed by
   the tests/benches. Cargo initially creates this directory but doesn't
   manage its content in any way, this is the responsibility of the test code.
+* `CARGO_WORKSPACE_DIR` --- Only set when building [integration test] or benchmark code.
+  This is a path to the workspace directory of the current package. If the current package
+  is not part of a workspace, it is a path to the directory containing the manifest of the package.
+  For a published registry crate it is a path to the workspace directory
+  of the package that (possibly transitively) imported that published crate.
```

Comment:
With the "for a published", it is making it sound like this is special cased when it isn't.

Also, "the path" would be better than "a path".

Maybe something like

"This is the path to the workspace directory of the package being built.  When there is no associated workspace, which includes all registry packages, this will be the directory contains the manifest of the package.